### PR TITLE
Add `buildStaticAddonCard()` to `addons-frontend-blog-utils`

### LINKED ISSUE #10297: Expose `buildStaticAddonCard` in `addons-frontend-blog-utils`
Once we have the library set up, we'll need a function to build a static add-on card.

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #10297 

---

This patch adds a new function to the `blog-utils` library, which allows to build static add-on cards (for the blog). The `dev` command for this library will start a small server to serve an HTML page for development purposes.

I'll add the rest of the UI (ratings, number of users) in a different patch. Same for the cosmetic changes to have the same look&feel as in the mocks (which weren't finalized at the time of writing).

## Screenshots

Note: buttons are blue now

Screen Shot 2021-03-32 at 11 22 14

Screen Shot 2021-03-32 at 11 22 53

Screen Shot 2021-03-32 at 11 22 44

### CODE REVIEW

File: README.md
Comment (eviljeff) [inline README.md:348]: #team-oxford-comma

File: package.json
Comment (eviljeff) [inline package.json:13]: I'm probably missing something about how `concurrently` works (I assumed from the naming it runs the commands simultaneously), but how can we serve blog-utils before we're finished building it?

File: src/blog-utils/StaticAddonCard/index.js
Comment (eviljeff) [inline src/blog-utils/StaticAddonCard/index.js:24]: Assuming there isn't some switching around in the frontend reducer, etc, `summary` is the field we extract from the manifest - and force to have a value before a version can be submitted for listed approval - while `description` is optional, so it's unlikely an add-on will have a description but not a summary.

File: src/blog-utils/StaticAddonCard/index.js
Comment (eviljeff) [inline src/blog-utils/StaticAddonCard/index.js:41]: this is a deliberately more restrictive list of tags, I guess.  (the sanitized summary can have a wider number of tags)

File: src/blog-utils/index.html
Comment (eviljeff) [inline src/blog-utils/index.html:10]: 😱 external files!

Comment (eviljeff) [reply src/blog-utils/index.html:10]: I note the "FOR DEVELOPMENT PURPOSES ONLY" at the top but these would be blocked by the CSP?

File: src/blog-utils/index.js
Comment (eviljeff) [inline src/blog-utils/index.js:19]: should this be in config so it can be overridden to point to a local/dev/stage url?

**Review by eviljeff — APPROVED** [review]

File: package.json
Comment (willdurand) [reply package.json:13]: The node server serves the dist files. If you browse the dev page as soon as it is up, the files will fail to load in the browser until these files are generated by webpack.

File: src/blog-utils/StaticAddonCard/index.js
Comment (willdurand) [reply src/blog-utils/StaticAddonCard/index.js:24]: Interesting, I wonder why we have that on AMO then. I will file an issue to investigate.

File: src/blog-utils/StaticAddonCard/index.js
Comment (willdurand) [reply src/blog-utils/StaticAddonCard/index.js:41]: I thought the summary could only have those two html tags because it is a "autolink field"? I think we had an issue about that because Fenix wasn't allowing any html tags at all.

Note: this list of tags comes from AMO, I only copied it.

File: src/blog-utils/index.html
Comment (willdurand) [reply src/blog-utils/index.html:10]: It's for local dev, served by a standalone node server so it won't be blocked by anything. This page isn't bundled in the lib.

File: src/blog-utils/StaticAddonCard/index.js
Comment (eviljeff) [reply src/blog-utils/StaticAddonCard/index.js:41]: Oh.  I looked at the list of tags for the underlying TranslatedField and it had a longer list, including the standard formatting ones, but it could easily have been overridden somewhere else.  Where did you get/copy the list from?

File: src/blog-utils/index.js
Comment (willdurand) [reply src/blog-utils/index.js:19]: I'd like to try to NOT bundle the frontend config in the lib if possible, which is why it's the prod url only. Also, I don't know exactly how deployments will work in addons-blog, so this might change..

File: src/blog-utils/StaticAddonCard/index.js
Comment (willdurand) [reply src/blog-utils/StaticAddonCard/index.js:24]: It looks like we're doing this for themes: https://github.com/mozilla/addons-frontend/blob/cabfc342cdbd1fed6c9907051cdedceb8998971a/src/amo/pages/Addon/index.js#L422-L424

File: src/blog-utils/StaticAddonCard/index.js
Comment (willdurand) [reply src/blog-utils/StaticAddonCard/index.js:41]: This is the issue I mentioned above: https://github.com/mozilla/addons-server/issues/15669

The list and this code actually comes from: https://github.com/mozilla/addons-frontend/blob/cabfc342cdbd1fed6c9907051cdedceb8998971a/src/amo/pages/Addon/index.js#L427-L430

File: src/blog-utils/StaticAddonCard/index.js
Comment (eviljeff) [reply src/blog-utils/StaticAddonCard/index.js:24]: iirc that was a LWT thing.  Static themes follow the same rules as extensions.

File: src/blog-utils/StaticAddonCard/index.js
Comment (eviljeff) [reply src/blog-utils/StaticAddonCard/index.js:41]: you're right.   summary is a `LinkifiedTranslation` rather than `PurifiedTranslation`, and it does only support `a` tags (I guess `br` are added somewhere along the way from converting `\n` to `<br>`)

**Pull request merged**
